spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -459,6 +459,7 @@ protected void actualStart() throws Exception {
 								}
 								this.logger.error("Error creating consumer; retrying in " + nextBackOff, e);
 								doShutdown();
+								this.consumersToRestart.clear();
```

Comment:
Hm. Looking to what you've done and what we have discussed before, plus the current logic, I'd say that you were right: we shouldn't add to `consumersToRestart` if we are not in the `DirectMessageListenerContainer.this.started`.
Just because this `while (!DirectMessageListenerContainer.this.started && isRunning()) {` are not going to exit until `DirectMessageListenerContainer.this.started = true;`.

WDYT?

Good catch though!
I like your brilliant vision 😄